packages/cli/src/credentials/credentials.types.ts

```diff
@@ -0,0 +1,13 @@
+import type { ICredentialsDb } from '../Interfaces';
+
+export interface CredentialWithSharings extends ICredentialsDb {
+	ownedBy?: UserSharingsDetails | null;
+	sharedWith?: UserSharingsDetails[];
+}
+
+export type UserSharingsDetails = {
+	id: string;
+	email: string;
+	firstName: string;
+	lastName: string;
+};
```

Comment:
Isn't it best if we base it off `User`? Omit some keys or just use parts of it? I'd like to keep a reference to the original object so if for any reason one changes the other is also somewhat reflected.

Reply:
created a new interface `IUser` in the workflow package. `User` implements that now.